src/zarr/v3/group.py

```diff
@@ -271,8 +289,32 @@ def __repr__(self):
     async def nchildren(self) -> int:
         raise NotImplementedError
 
-    async def children(self) -> AsyncIterator[AsyncArray, AsyncGroup]:
-        raise NotImplementedError
+    async def children(self) -> AsyncGenerator[AsyncArray, AsyncGroup]:
+        """
+        Returns an AsyncGenerator over the arrays and groups contained in this group.
+        This method requires that `store_path.store` supports directory listing.
+        """
+        if not self.store_path.store.supports_listing:
+            msg = (
+                f"The store associated with this group ({type(self.store_path.store)}) "
+                "does not support listing, "
+                "specifically via the `list_dir` method. "
+                "This function requires a store that supports listing."
+            )
+
+            raise ValueError(msg)
+        subkeys = await self.store_path.store.list_dir(self.store_path.path)
+        # would be nice to make these special keys accessible programmatically,
+        # and scoped to specific zarr versions
+        subkeys_filtered = filter(lambda v: v not in ("zarr.json", ".zgroup", ".zattrs"), subkeys)
```

Comment:
I'm still confused here. Don't we want to be looking for `.zarray` to find v2 arrays?

Reply:
if `foo/bar/` is the prefix for a group, then we might expect to find either v2 group stuff (`foo/bar/.zattrs`, `foo/bar/.zgroup`) or v3 group stuff (`foo/bar/zarr.json`), but `foo/bar/.zarray` would be unexpected, since we know `foo/bar` is a group, and the same path can't correspond to both a group and an array.